### PR TITLE
EMSUSD-632 prevent root metadata changes when not targeting root

### PR DESCRIPTION
**Pull request by pierrebai-adsk**
- Add a applyRootLayerMetadataRestriction helper function to enforce not modifying the root layer metadata when not targeting the root layer.
- The only restriction applied is about the default prim.
- Call it from within the applyCommandRestriction helper function.
- Call it from within the default prim commands.
- Add unit tests for the rename, parent and group default prim restrictions.
- Add tests for the default prim commands.

### CODE REVIEW

**Comment by pierrebai-adsk** [comment]
The Attribute Editor now uses the command and can be undone.

**Comment by pierrebai-adsk** [comment]
Once again, the only PF failure is one interactive test in a single PF where a USD file was left behind in the temp folder from e previous PF, that failure can be ignored.

File: plugin/adsk/scripts/AETemplateHelpers.py [hunk 1]
```diff
@@ -47,19 +50,26 @@ def GetDefaultPrimName(proxyShape):
...
+            return False
+        
+        ufeCmd.execute(cmd)
+        return True
     except Exception as e:
-        debugMessage('SetDefaultPrim() - Error: %s' % str(e))
-        pass
-    return ''
+        # Note: we do want to tell the user why the set or clear failed.
+        OpenMaya.MGlobal.displayError((str(e)))
```
Comment (seando-adsk) [inline plugin/adsk/scripts/AETemplateHelpers.py:71]: Not a big deal, but isn't there an extra set of () here?

File: test/lib/ufe/testDefaultPrimCmds.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+
+#
+# Copyright 2020 Autodesk
```
Comment (seando-adsk) [inline test/lib/ufe/testDefaultPrimCmds.py:4]: Should be current year.

**Review by seando-adsk — CHANGES_REQUESTED** [review]
Great work. Just a couple of very minor comments.

**Review by seando-adsk — APPROVED** [review]

**Pull request merged**